lib/ClientConnectContext.php

```diff
@@ -52,6 +54,21 @@ public function getMaxAttempts(): int {
         return $this->maxAttempts;
     }
 
+    public function withDnsTypeRestriction($type): self {
```

Comment:
I think this should be `int $type = null`. Not specifying `= null` prevents a BC break once we use nullable types, but `withBindTo` requires that anyway.

Reply:
fair enough :)

Reply:
Although that would mean you can't actually pass null in as the parameter, doesn't it. To reset the type - which, admittedly, you are unlikely to want to do, you'd need to call the method without any parameters which seems a bit odd.

Reply:
No, `= null` automatically adds nullability.